ledger/cow.go

```diff
@@ -285,3 +290,34 @@ func (cb *roundCowState) commitToParent() {
 func (cb *roundCowState) modifiedAccounts() []basics.Address {
 	return cb.mods.Accts.ModifiedAccounts()
 }
+
+// CalculateTotals calculates the totals given the changes in the StateDelta.
+// these changes allow the validator to validate that the totals still align with the
+// expected values. ( i.e. total amount of algos in the system should remain consistent )
+func (cb *roundCowState) CalculateTotals() error {
+	// this method applies only for the top level roundCowState
+	if cb.commitParent != nil {
+		return nil
+	}
+	totals := cb.prevTotals
+	var ot basics.OverflowTracker
+	totals.ApplyRewards(cb.mods.Hdr.RewardsLevel, &ot)
+
+	for i := 0; i < cb.mods.Accts.Len(); i++ {
+		accountAddr, updatedAccountData := cb.mods.Accts.GetByIdx(i)
+		previousAccountData, lookupError := cb.lookupParent.lookup(accountAddr)
+		if lookupError != nil {
+			return fmt.Errorf("roundCowState.CalculateTotals unable to load account data for address %v", accountAddr)
+		}
+		totals.DelAccount(cb.proto, previousAccountData, &ot)
+		totals.AddAccount(cb.proto, updatedAccountData, &ot)
+	}
+	cb.mods.Totals = totals
```

Comment:
Maybe this should be called after the error checks so that `cb` is not be modified if returning an error?

Reply:
makes sense.